### PR TITLE
feature: Added CI to verify proper formatting of PR titles.

### PR DESCRIPTION
**Pull request by ayushrakesh**
Added one CI to check that PR titles are properly formatted.

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Added one .yaml file for CI.

## Verification

make test

### CODE REVIEW

File: .github/workflows/pr-title-check.yml
Comment (douglascamata) [inline .github/workflows/pr-title-check.yml:26]: This is still wrong. Please reread the pull request template. The allowed prefix is not only `query:` and `.*:` (not that `.*` here isn't a regex it means "something that applies to all components).

**Review by douglascamata — CHANGES_REQUESTED** [review]
The PR title format's still wrong to me. 

Plus now we have two source of information regarding how PRs should be written. I would prefer to have one source and link them.

I'm also not sure about how helpful this linting is. It can block a PR from being merged when there are legitimate reasons to break the pattern. It's very easy and fast for any maintainer to edit the title of a PR before merging it. 

Curious to see what other maintainers have to say about this.

File: .github/workflows/pr-title-check.yml
Comment (ayushrakesh) [reply .github/workflows/pr-title-check.yml:26]: @douglascamata Okay, means "query" or some single word that applies to all components right? If it is then can you suggest what could be best possible word?

Comment (douglascamata) [reply .github/workflows/pr-title-check.yml:26]: The PR title, as the pull request template says, should start with the name of the Thanos component involved in the PR. We have a list of the components in the official docs (https://thanos.io/tip/components/). 

You can also check the history of PRs. You might see some that affected one or more components at the same time. For sure some that affected all of them too.

Comment (ayushrakesh) [reply .github/workflows/pr-title-check.yml:26]: PREFIXES="query:|compactor:|query-frontend:|receiver:|rule:|sidecar:|store:|tools:"
 
 @douglascamata Is this exactly correct?

Comment (douglascamata) [reply .github/workflows/pr-title-check.yml:26]: You're missing the `.*` (string) prefix, which has to be escaped to not be confused with the meaning that `.*` has in regexes.

File: docs/components/tools.md
Comment (douglascamata) [inline docs/components/tools.md:1]: These changes are just noise. Same applies to many other files the same folder.

File: CONTRIBUTING.md
Comment (douglascamata) [inline CONTRIBUTING.md:205]: What about multi-component PRs like https://github.com/thanos-io/thanos/pull/7406?

What about a PR like this one itself?

**Review by douglascamata — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md
Comment (ayushrakesh) [reply CONTRIBUTING.md:205]: @douglascamata What about this [docs/components/tools.md](https://github.com/thanos-io/thanos/pull/7470/files/4d41d7036df1de28dc40ca5553525102da7e36e9#diff-89840bcebb29ac15c25fad738d4d95c327074e0559da78e4851b696d9533c267) is this related to my work? 

For PRs involving multiple components, maybe we can use syntax like this -> `query,store: Enhance data retrieval efficiency`. What you say?

Comment (douglascamata) [reply CONTRIBUTING.md:205]: Seems okay to me. But mind that I'm not a maintainer, so my review is not conclusive. A maintainer will still have to review this.

**Pull request closed without merging**
